### ISSUE TITLE
dropout when eval model

### ISSUE THREAD

**Issue by boundles** [Eromera/erfnet_pytorch#6]
Hi, did you use dropout when eval model? I use Caffe to reproduce ERFNet, but  it costs about 100ms on forward. Could you give me some suggestions?

**Comment by Eromera** [comment]
Hi! Thanks for the message. Dropout in any framework is normally only used in train mode, in eval mode there is no use for it since it consists on deactivating some weights while training but it has no effect while doing forward in eval mode.

I think Caffe is less efficient that PyTorch but I am not sure if that alone is increasing from the 25-30 ms I get in PyTorch to the 100ms you get in Caffe.

What CUDA version are you using? Are you using CuDNN? (that should accelerate many networks like ERFNet). Also you can get a boost if you remove the Batch Normalization layers but for that you would need to "absorb" the mean and std saved in the batch-norm layers to the weights of the previous conv layer. There should be code available to do that, you could look for "batch normalization absorber" or "BN absorber" or something like that, there is code in Torch but I don't know about Caffe.

Also what resolution are you using? 25-30 ms is in 1024x512, in 2048x1024 it should be 4x which would make sense with those 100ms.

**Comment by boundles** [comment]
My test code has some bugs so it's slower. Thanks for your reply.
